Hazel/src/Hazel/Core/Math/degrees.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+
+
+#include<type_traits>
+#include "Hazel/Core/Base.h"
+
+
+
+// Extensive use of explicit for all converstion reduces the chances of unintentionally copying
+//values between radians and degrees without performing a conversion calculation.
+// implicit conversions are the devil's playground.
+
+
+
+namespace Hazel
+{
+	class Radians;
+
+	class Degrees
+	{
+	public:
+		constexpr Degrees( void ) noexcept = default;
+		~Degrees( void ) noexcept = default;
+
+		constexpr Degrees( const Degrees& other ) noexcept : m_Angle( other.m_Angle ) {}
+		explicit Degrees( const Radians& radians ) noexcept;
+		Degrees( Degrees&& other ) noexcept { std::swap( this->m_Angle, other.m_Angle ); }
```

Comment:
Implementing a move constructer for a float seems a bit overkill.

Reply:
"= default" works with it in Visual Studio 2019 but I wasn't sure about other compilers.   Rule of Five, if you implement one constructor, you should implement them all--obviously it's just a guideline and by itself it doesn't make a big performance difference especially in this case but they're trivial to implement if you do it as you're going but a pain in the ____ to go back and do later.  Having them gives the compiler more opportunities for optimization so I think it's wise to just throw them as you go.

Also, since the creation, itself, of Hazel is meant be educational; creating move ctors for simple things like this can be informative for newer programmers later on with larger classes.